src/ember-app.js

```diff
@@ -492,20 +492,30 @@ class EmberApp {
       }
     }
 
-    debug('reading array of app file paths from manifest');
-    let appFiles = manifest.appFiles.map(function(appFile) {
-      return path.join(distPath, appFile);
-    });
+    let appFiles, vendorFiles, htmlFile, html;
 
-    debug('reading array of vendor file paths from manifest');
-    let vendorFiles = manifest.vendorFiles.map(function(vendorFile) {
-      return path.join(distPath, vendorFile);
-    });
+    if (manifest.htmlEntrypoint) {
+      let htmlEntrypoint = require('./html-entrypoint');
+      ({ appFiles, vendorFiles, html } = htmlEntrypoint(distPath, manifest.htmlEntrypoint));
+    } else {
+      debug('reading array of app file paths from manifest');
+      appFiles = manifest.appFiles.map(function(appFile) {
+        return path.join(distPath, appFile);
+      });
+
+      debug('reading array of vendor file paths from manifest');
+      vendorFiles = manifest.vendorFiles.map(function(vendorFile) {
+        return path.join(distPath, vendorFile);
+      });
+
+      htmlFile = path.join(distPath, manifest.htmlFile);
+    }
```

Comment:
Seems like this should actually be incrementing the schema version ([see here](https://github.com/ember-fastboot/fastboot/blob/master/src/fastboot-schema-versions.js)) as the opt-in and we "upgrade" to the latest version automatically. That reduces overall branching in this code, and means that `fastboot` itself can always assume the same thing.

Reply:
If you are ready to move all apps to this new format, then I agree and we can make this v5.  I was not being so forward as to suggest that yet. At the moment it is an alternative, not an upgrade.

Making it an upgrade would certainly improve the error message people get if they try to use embroider without updating to this version of fastboot.

As for less branching: I realize that's the goal, but I don't actually agree that the current schema versioning system achieves it. Making this new format v5 would not result in fewer branches.

Reply:
> If you are ready to move all apps to this new format, then I agree and we can make this v5. I was not being so forward as to suggest that yet. At the moment it is an alternative, not an upgrade.

I see no reason to keep two long running branches here, there should be one logical successor with the other path being planned for deprecation/removal.

AFAICT, there really isn't a reason that ember-cli-fastboot can't emit `htmlEntrypoint` instead of its current system reducing yet another difference between Embroider and classic build worlds...

> As for less branching: I realize that's the goal, but I don't actually agree that the current schema versioning system achieves it.

Ya, not sure I agree, but that isn't really the point. The goal of the system is to ensure we only deal with grabbing information from the manifest in a single place **and** we know what kind of thing we are dealing with. As it stands, you could easily have an `htmlManifest` **and** `vendorFiles` and have no clue why things don't work as you expect.

Reply:
+1 if we think we have settled on `htmlEntrypoint` being the API, it would be ideal for `ember-cli-fastboot` to adopt it as well and reduce the clutter of maintaining between the two build systems.

Reply:
> it would be ideal for ember-cli-fastboot to adopt it as well and reduce the clutter of maintaining between the two build systems

Ya, agree. Once we land this, we should make a tracking issue over on ember-cli-fastboot.